gremlin-go/driver/connection_test.go

```diff
@@ -1,18 +1,117 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
 package gremlingo
 
 import (
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/assert"
 	"golang.org/x/text/language"
 )
 
 const runIntegration = true
+const testHost string = "localhost"
+const testPort int = 8182
 
 func TestConnection(t *testing.T) {
+	t.Run("Test g.V().count()", func(t *testing.T) {
+		if runIntegration {
+			remote, err := NewDriverRemoteConnection(testHost, testPort)
+			assert.Nil(t, err)
+			assert.NotNil(t, remote)
+			g := _Traversal().WithRemote(remote)
+
+			// Drop vertices that were added.
+			_, err = g.V().Drop().Iterate()
+			assert.Nil(t, err)
+
+			// Read count again, should be 0.
+			results, err := g.V().Count().ToList()
+			assert.Nil(t, err)
+			assert.NotNil(t, results)
+			assert.Equal(t, 1, len(results))
+			var count int32
+			count, err = results[0].GetInt32()
+			assert.Nil(t, err)
+			assert.Equal(t, int32(0), count)
+			time.Sleep(50 * time.Millisecond)
```

Comment:
What's the sleep for again? Does this rely on the server having a larger timeout?

Reply:
Because of the async nature of our API, in some cases the previous function traversal hadn't finished and this caused incorrect results. I can probably just make sure I consumed the results properly and remove the timeouts, let me look into that when I have some time.